File: Maximum and minimum of an array.py
# Time Complexity : O(N)
# Space Complexity : O(1)
# Did this code successfully run on Leetcode : yes
# Any problem you faced while coding this : Don't know whethter this is optimal

def min_and_max(arr):
    arr_size = len(arr)
    min = 0
    max = 0

    if arr_size == 1: 
            max = arr[0] 
            min = arr[0]

    if arr[0] > arr[1]: 
        max = arr[0] 
        min = arr[1] 
    else: 
        max = arr[1] 
        min = arr[0] 

    for i in range(2, arr_size): 
        if arr[i] > max: 
            max = arr[i] 
        elif arr[i] < min: 
            min = arr[i]
    print("Minimum element is", min) 
    print("Maximum element is", max) 
arr = [1000, 11, 445, 1, 330, 3000] 
min_and_max(arr)

File: Maximum_and_Minimum_of_an_Array.cpp
/*
Time Complexity: O(n)
Space Complexity: O(1)
Did this code successfully run on Leetcode :yes
Any problem you faced while coding this : no
*/
#include <bits/stdc++.h>
#include <iostream>

using namespace std;

    std::vector<int> maxmin(std::vector<int>& nums) {
        int mn = INT_MAX;
        int mx = INT_MIN;
        if(nums.size()==0){
            return {{0,0}};
        }
        for(int i = 0; i< nums.size()-1; i++){
            if(nums[i]>nums[i+1]){
                mn = min(mn,nums[i+1]);
                mx = max(mx,nums[i]);
            } else{
                mn = min(mn,nums[i]);
                mx = max(mx,nums[i+1]);
            }
        }
        return {mn,mx};
    }

int main(){
    std::vector<int> n = {9,3,4,6,2,8,1};
    std::vector<int> ans = maxmin(n);
    cout<<ans[0]<<" "<<ans[1];
    return 0;
}